projects/dsp-ui/src/lib/action/pipes/string-transformation/truncate.pipe.ts

```diff
@@ -0,0 +1,39 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+/**
+ * This pipe can be used to shorten long text by a defined length.
+ *
+ * In markup:
+ *
+ * {{ str | dspTruncate:['24'] }}
+ *
+ * or
+ *
+ * {{ str | dspTruncate:['24', '...'] }}
+ *
+ *
+ * The first parameter defines the length where to truncate the string.
+ * Second optional parameter defines the characters to append to the shortened string. Default is '...'.
+ *
+ * The advantage of this pipe over the default Angular slice pipe is the simplicity of adding
+ *  additional characters at the end of the shortened string.
+ * The same construct with Angular slice pipe looks as follow: `{{ (str.length>24)? (str | slice:0:24)+'...':(str) }}`.
+ *
+ */
+@Pipe({
+    name: 'dspTruncate'
+})
+export class TruncatePipe implements PipeTransform {
+
+    transform(value: string, args: string[]): string {
```

Comment:
Could't the second param be numeric?

Reply:
The second param needs to be a string because you can override the default truncation string with a custom string when you give a second element in the array. Like this:
`pipe.transform('The quick brown fox jumps over the lazy dog.', ['19', '!'])`

Reply:
Could rest params be used here? https://www.typescriptlang.org/docs/handbook/functions.html#rest-parameters